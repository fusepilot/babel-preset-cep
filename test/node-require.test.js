const preset = require('../src');
const babel = require('@babel/core');

it('replaces commented node-require require functions with window.cep_node.require', () => {
  const example = `
    // node-require
    const fs = require("fs")
  `;

  const { code } = babel.transform(example, { presets: [preset] });
  expect(code).toMatchSnapshot();
});

it('doesnt replace normal requires', () => {
  const example = `
    const fs = require("fs")
  `;

  const { code } = babel.transform(example, { presets: [preset] });
  expect(code).toMatchSnapshot();
});

it('non naive example', () => {
  const example = `
    const path = require("path")

    // node-require
    const fs = require("fs")

    // node-require
    const os = require("os")

    require("babel-preset")

    const util = require("util")

    // node-require
    const crypto = require("crypto")
  `;

  const { code } = babel.transform(example, { presets: [preset] });
  expect(code).toMatchSnapshot();
});
