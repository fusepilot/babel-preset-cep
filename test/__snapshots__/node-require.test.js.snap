// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesnt replace normal requires 1`] = `
"\\"use strict\\";

var fs = require(\\"fs\\");"
`;

exports[`non naive example 1`] = `
"\\"use strict\\";

var path = require(\\"path\\");

// node-require
var fs = (window.cep_node || window).require(\\"fs\\");

// node-require
var os = (window.cep_node || window).require(\\"os\\");

require(\\"babel-preset\\");

var util = require(\\"util\\");

// node-require
var crypto = (window.cep_node || window).require(\\"crypto\\");"
`;

exports[`replaces commented node-require require functions with window.cep_node.require 1`] = `
"\\"use strict\\";

// node-require
var fs = (window.cep_node || window).require(\\"fs\\");"
`;
